src/lib/components/Article.svelte

```diff
@@ -104,62 +92,54 @@
       window.removeEventListener('click', hideTocOnClick);
     };
   });
-
-  const converter = new showdown.Converter();
 </script>
 
-{#await getEvents(index)}
+{#await getContentRoot(index)}
   <Sidebar class='sidebar-leather fixed top-20 left-0 px-4 w-60'>
     <SidebarWrapper>
       <Skeleton/>
     </SidebarWrapper>
   </Sidebar>
   <TextPlaceholder class='max-w-2xl'/>
-{:then events}
-  {#if showTocButton && !showToc}
-    <Button
-      class='btn-leather fixed top-20 left-4 h-6 w-6'
-      outline={true}
-      on:click={ev => {
-        showToc = true;
-        ev.stopPropagation();
-      }}
-    >
-      <BookOutline />
-    </Button>
-    <Tooltip>
-      Show Table of Contents
-    </Tooltip>
-  {/if}
-  {#if showToc}
-    <Sidebar class='sidebar-leather fixed top-20 left-0 px-4 w-60' {activeHash}>
-      <SidebarWrapper>
-        <SidebarGroup class='sidebar-group-leather overflow-y-scroll'>
-          {#each events as event}
-            <SidebarItem
-              class='sidebar-item-leather'
-              label={event.getMatchingTags('title')[0][1]}
-              href={`${$page.url.pathname}#${normalizeHashPath(event.getMatchingTags('title')[0][1])}`}
-            />
-          {/each}
-        </SidebarGroup>
-      </SidebarWrapper>
-    </Sidebar>
+{:then rootId}
+  {#if rootId}
+    {#if showTocButton && !showToc}
+      <Button
+        class='btn-leather fixed top-20 left-4 h-6 w-6'
+        outline={true}
+        on:click={ev => {
+          showToc = true;
+          ev.stopPropagation();
+        }}
+      >
+        <BookOutline />
+      </Button>
+      <Tooltip>
+        Show Table of Contents
+      </Tooltip>
+    {/if}
+    <!-- TODO: Get TOC from parser. -->
+    <!-- {#if showToc}
+      <Sidebar class='sidebar-leather fixed top-20 left-0 px-4 w-60' {activeHash}>
+        <SidebarWrapper>
+          <SidebarGroup class='sidebar-group-leather overflow-y-scroll'>
+            {#each events as event}
+              <SidebarItem
+                class='sidebar-item-leather'
+                label={event.getMatchingTags('title')[0][1]}
+                href={`${$page.url.pathname}#${normalizeHashPath(event.getMatchingTags('title')[0][1])}`}
+              />
+            {/each}
+          </SidebarGroup>
+        </SidebarWrapper>
+      </Sidebar>
+    {/if} -->
+    <div class='flex flex-col space-y-4 max-w-2xl'>
+      <Preview rootId={rootId} />
+    </div>
+  {:else}
+    <!-- TODO: Display empty state. -->
```

Comment:
Ah, I see, now. Ignore my previous comment about the ToC.

Reply:
Yes we'll tackle ToC with [ALEX-50](https://shadowysupercoding.atlassian.net/browse/ALEX-50?atlOrigin=eyJpIjoiMDI3MmRmMTcyZDMyNGYyMjhiMzVjNWQ5OTMwZWU0YjkiLCJwIjoiaiJ9).